### PR TITLE
processes plugin: Code cleanup + add per-process open files count

### PR DESCRIPTION
**Pull request by rpv-tomsk**
The `procstat_entry_s` used in improper way.
Added new type and simplified the code - removed unneeded assignments and values copying.

### CODE REVIEW

**Comment by rpv-tomsk** [comment]
That is too hard to make changes in this plugin. There is too much unnecessary actions.

What do you think, @octo?

**Comment by rpv-tomsk** [comment]
The second patch tries to fix gaps, caused by terminated processes.
When process terminates, huge value is decreased from counter, causing negative derived value.

**Comment by rpv-tomsk** [comment]
Added commit here for 'Add per-process open files count #836'.

**Comment by rpv-tomsk** [comment]
After I had put this on my production, I found a drop on one of charts (Updated at 07:33):



Avg value before update was around ~65 and around ~40 after upgrade.
I have checked this with `# while true; do cat /proc/2317/stat && sleep 10; done` and "spreadsheet" and found what current value is right. Before upgrade there was collectd-5.5.2 code running.
That change is only at one chart. I have two mysql instances monitored same way and same collectd process, and there is not such drop or rise. Quite strange.

**Comment by rpv-tomsk** [comment]
Changed `files` to `file_handles`.

Work finished, please review/merge it.

**Comment by rpv-tomsk** [comment]
@rubenk, Can you please review this?

**Comment by rpv-tomsk** [comment]
ping

**Comment by Mogztter** [comment]
This feature would be really useful to us.
I can't review the code because I don't know C but let me know if you need help in order to get it merged.

**Comment by rpv-tomsk** [comment]
Hi!

I'm just using this (like other my PR) in my production environment, so I don't experience any problems with code quality. 

Thanks for your attention to this patch request. 
I will update it to fix merge conflict, then you can try to get it merged.

**Comment by rpv-tomsk** [comment]
Rebased 

Taken the src/processes.c file from cb8683ce442a0e7c14b910a7fff30a71b1acef35, 36cfb411f16ed11dadf70619f9779bf47c363340, 7c33e5d08e5567251d0a16a18bf4fd99b36f1ad9 commits, passed throught clang-format and done new commits.

**Comment by rubenk** [comment]
This is Linux-only right? If so, please add that fact to the docs and commit message.

**Comment by rpv-tomsk** [comment]
Hi Ruben!

That change would be enough?

**Comment by rubenk** [comment]
Enough for what?

**Comment by rpv-tomsk** [comment]
For this: 

> This is Linux-only right? If so, please add that fact to the docs and commit message.

File: src/processes.c [hunk 1]
```diff
@@ -512,6 +562,8 @@ static int ps_config(oconfig_item_t *ci) {
   const size_t max_procname_len = MAXCOMLEN - 1;
 #endif
 
+  procstat_t *ps;
```
Comment (rubenk) [inline src/processes.c]: perhaps move this to line 587, so it becomes `procstat_t *ps = ps_list_register(c->values[0].value.string, NULL);`?

**Comment by rubenk** [comment]
Ah, I didn't see your change. Github lag I guess. Yes, it is fine.
Thanks Pavel!

**Comment by Mogztter** [comment]
Awesome, thanks @rpv-tomsk and @rubenk

**Pull request merged**